src/Layers/xrRender/BufferUtils.h

```diff
@@ -0,0 +1,116 @@
+#pragma once
+
+u32 GetFVFVertexSize(u32 FVF);
+u32 GetDeclVertexSize(const VertexElement* decl, DWORD Stream);
+u32 GetDeclLength(const VertexElement* decl);
+
+struct SDeclaration;
+
+void ConvertVertexDeclaration(u32 FVF, SDeclaration* decl);
+void ConvertVertexDeclaration(const VertexElement* dxdecl, SDeclaration* decl);
+void ConvertVertexDeclaration(const xr_vector<VertexElement>& declIn, xr_vector<InputElementDesc>& declOut);
+
+namespace BufferUtils
+{
+HRESULT CreateVertexBuffer(VertexBufferHandle* pBuffer, const void* pData, UINT DataSize, bool bImmutable = true);
+HRESULT CreateIndexBuffer(IndexBufferHandle* pBuffer, const void* pData, UINT DataSize, bool bImmutable = true);
+HRESULT CreateConstantBuffer(ConstantBufferHandle* ppBuffer, UINT DataSize);
+};
+
+class VertexStagingBuffer
+{
+public:
+    VertexStagingBuffer();
+    ~VertexStagingBuffer();
+
+    void Create(size_t size);
+    void Destroy();
+
+    bool IsValid() const;
+    void* GetHostPointer();
+    VertexBufferHandle GetBufferHandle() const;
+    void Flush();
```

Comment:
It is better to rename this function to `Unlock`.
```suggestion
    void Unlock();
```